packages/twenty-front/src/modules/workspace/components/WorkspaceProviderEffect.tsx

```diff
@@ -24,6 +33,31 @@ export const WorkspaceProviderEffect = () => {
 
   const isMultiWorkspaceEnabled = useRecoilValue(isMultiWorkspaceEnabledState);
 
+  useEffect(() => {
+    if (isDefaultDomain || !getPublicWorkspaceData) return;
+
+    // checks if any non-SSO providers are enabled
+    if (
+      Object.values(getPublicWorkspaceData.authProviders).filter(
+        (value) => typeof value === 'boolean' && value,
+      ).length !== 0
+    )
+      return;
+
+    if (getPublicWorkspaceData.authProviders.sso.length > 1) {
+      return setSignInUpStep(SignInUpStep.SSOIdentityProviderSelection);
+    }
+
+    if (getPublicWorkspaceData.authProviders.sso.length === 1) {
+      redirectToSSOLoginPage(getPublicWorkspaceData.authProviders.sso[0].id);
```

Comment:
redirections should be made in PageChangeEffect :)

Reply:
actually this is a window.location redirection, it's fine :)

However, this should be part of the SignInUpWorkspaceScopeFormEffect, let's centralize the logic